linux/src/threadpool_linux.c

```diff
@@ -325,18 +297,26 @@ static int reallocate_threadpool_array(THREADPOOL* threadpool)
 
 static void threadpool_dispose(THREADPOOL* threadpool)
 {
-    /* Codes_SRS_THREADPOOL_LINUX_07_013: [ threadpool_destroy shall perform an implicit close if threadpool is open. ]*/
-    internal_close(threadpool);
+    /* Codes_SRS_THREADPOOL_LINUX_07_089: [ threadpool_dispose shall signal all threads that the threadpool is closing by setting the flag to stop the threads to 1. ]*/
+    (void)InterlockedHL_SetAndWakeAll(&threadpool->stop_thread, 1);
+    for (int32_t index = 0; index < threadpool->used_thread_count; index++)
+    {
+        int dont_care;
+        /* Codes_SRS_THREADPOOL_LINUX_07_027: [ threadpool_dispose shall join all threads in the threadpool. ]*/
+        if (ThreadAPI_Join(threadpool->thread_handle_array[index], &dont_care) != THREADAPI_OK)
+        {
+            LogError("Failure joining thread number %" PRId32 "", index);
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

We usually have the other branch in the code even when emoty. We put a comment in the empty branch like "// everything ok". That is only to make it nicer and explicit to developers not to wonder that a branch was missed.

Reply:
Added an else block